tests/components/vicare/conftest.py

```diff
@@ -57,7 +78,9 @@ async def mock_vicare_gas_boiler(
     hass: HomeAssistant, mock_config_entry: MockConfigEntry
 ) -> AsyncGenerator[MockConfigEntry, None]:
     """Return a mocked ViCare API representing a single gas boiler device."""
-    fixtures = ["vicare/Vitodens300W.json"]
+    fixtures: list[tuple[list[str], str]] = [
+        (["type:boiler"], "vicare/Vitodens300W.json")
```

Comment:
```suggestion
        ({"type:boiler"}, "vicare/Vitodens300W.json")
```

We should create a set from the beginning and not converting it on each call in `hasRoles`